service-base/src/main/java/org/eclipse/hono/service/AbstractProtocolAdapterBase.java

```diff
@@ -1086,6 +1087,53 @@ protected final Message newMessage(
             final JsonObject registrationInfo,
             final Integer timeUntilDisconnect) {
 
+        return newMessage(target, publishAddress, contentType, payload, tenant, registrationInfo, timeUntilDisconnect,
+                null);
+    }
+
+    /**
+     * Creates a new AMQP 1.0 message.
+     * <p>
+     * Subclasses are encouraged to use this method for creating {@code Message} instances to be sent downstream in
+     * order to have required Hono specific properties being set on the message automatically.
+     * <p>
+     * This method creates a new {@code Message} and sets
+     * <ul>
+     * <li>its <em>content-type</em> to the given value,</li>
+     * <li>its <em>creation-time</em> to the current system time and</li>
+     * <li>its payload as an AMQP <em>Data</em> section</li>
+     * </ul>
+     * The message is then passed to {@link #addProperties(Message, ResourceIdentifier, String, TenantObject, JsonObject, Integer)}.
+     *
+     * @param target The resource that the message is targeted at.
+     * @param publishAddress The address that the message has been published to originally by the device. (may be
+     *            {@code null}).
+     *            <p>
+     *            This address will be transport protocol specific, e.g. an HTTP based adapter will probably use URIs
+     *            here whereas an MQTT based adapter might use the MQTT message's topic.
+     * @param contentType The content type describing the message's payload or {@code null} if no content type
+     *                    should be set.
+     * @param payload The message payload.
+     * @param tenant The tenant that the device belongs to.
+     * @param registrationInfo The device's registration information as retrieved by the <em>Device Registration</em>
+     *            service's <em>assert Device Registration</em> operation.
+     * @param timeUntilDisconnect The number of milliseconds until the device that has published the message
```

Comment:
IMHO this is *seconds*, not *millis*.

Reply:
My bad. I have copied the javadoc of the existing `newMessage(...)` method and it describes `timeUntilDisconnect` as `The number of milliseconds ...`. It is an already existing bug and I will correct in the already existing  `newMessage(...)` method too.